content/ember-v4-typescript-support-update.md

```diff
@@ -0,0 +1,108 @@
+
+2. **If you’re a TypeScript Ember user**, you can now use v4 in your `@types` dependencies. That will give you the same benefits as JavaScript users get via the up-to-date definitions for Ember itself, but it will also provide *much* better feedback and errors in many cases, because we are now using the latest and greatest features of TypeScript for these types.
+
+Note that we have *not* yet removed the v3 deprecations or updated the strictness of the Ember Data v4 types, but we *have* updated Ember Data’s types to use the same minimum TypeScript version and to be compatible with the changes made in Ember’s types. (Unfortunately, none of the Typed Ember maintainers has deep familiarity with Ember Data!) If you’d like to help get those up to date, reach out to us in [#e-typescript on the Ember Discord][e-typescript]!
+
+
+## Notable changes
+
+* The types now use `"strict": true` in their compiler options. *In general*, this will not cause you any work; instead, it means that internally the types have to be more robust, which should cause you *fewer* issues when working in strict mode yourself. However, in a few cases, it meant we caught type safety bugs during the upgrade, which may show up in changes like those listed below.
+
+* **Breaking:** Throughout, the types use `unknown` in many places where they previously used `any`.
+
+* **Breaking:** many of the API types have been improved and many bugs fixed. This means that they are likely to reject a number of invocations which were allowed previously, but which would produce bugs at runtime, were misleading or confusing, etc.
+
+    A few notable examples:
+
+    * The `beforeModel` and `afterModel` hooks on `Route`s have been updated to require that you return either a `Promise` or *nothing*, since return values are otherwise ignored. If you were returning non-`Promise` values previously, you should remove those `return` statements.
+
+    <!-- alex ignore invalid -->
+    * The `@ember/runloop` functions all now robustly check their arguments. If you were passing invalid arguments for the callbacks, the compiler will  now catch that.
+
+<!-- alex ignore just -->
+* **Breaking:** The `TestContext` for the callbacks used in `module`, `beforeEach`, `afterEach`, `test`, etc. has moved from `ember-test-helpers` to `@ember/test-helpers`. Having nearly everything be in `@ember/test-helpers` but *not* `TestContext` has been a persistent source of confusion over the years, and was just a bit of historical baggage left over from before `@ember/test-helpers` existed. (If [RFC #0785](https://github.com/emberjs/rfcs/pull/785) is merged, we will also be able to move away from using the `TestContext` at all!)
+
+
+## Upgrading
+
+To upgrade to using these types:
```

Comment:
This reads funny, I thought there were missing words at first 😅 I don't really have a good suggestion though.

```suggestion
To use these types, follow these upgrade steps:
```

?